modules/httpcache/httpcache.go

```diff
@@ -31,7 +31,7 @@ func AddCacheControlToHeader(h http.Header, maxAge time.Duration, additionalDire
 		directives = append(directives, "max-age=0", "private", "must-revalidate")
 
 		// to remind users they are using non-prod setting.
-		h.Add("X-Gitea-Debug", "RUN_MODE="+setting.RunMode)
+		h.Set("X-Gitea-Debug", "RUN_MODE="+setting.RunMode)
```

Comment:
Why not clean the incorrect calls?

This `Add` is somewhat done by purpose, to see how many calls to this function. And there may be other "Debug Headers" in the future, I just do not want to overwrite other headers.

Reply:
The three headers are added in the following order:
1. https://github.com/go-gitea/gitea/blob/4c59c8c7682da31410decba3bd868fde5116e073/modules/context/context.go#L756
2. https://github.com/go-gitea/gitea/blob/4c59c8c7682da31410decba3bd868fde5116e073/modules/httpcache/httpcache.go#L128
3. https://github.com/go-gitea/gitea/blob/4c59c8c7682da31410decba3bd868fde5116e073/modules/context/context.go#L391

How do you suggest to clean it?

Reply:
Surprise .... actually there is already a conflict if I understand correctly: other `AddCacheControlToHeader` calls would drop the `"no-transform"` added before.

I think we need some new approaches for the cache header management.

Reply:
To make the problem clear (instead of ignoring it) and remove duplicate header output and still remind future developers, I think a fix could be:

1. move the `Add("X-Gitea-Debug", "RUN_MODE="+setting.RunMode)` to `context.go` , only be called once.
2. use `Add("X-Gitea-Debug", "CacheControl=" + args ...)` in `AddCacheControlToHeader`.

Then, the future developers would see 4 X-Gitea-Debug headers like:

```
X-Gitea-Debug: RUN_MODE=dev
X-Gitea-Debug: CacheControl=0, no-transform
X-Gitea-Debug: CacheControl=300s
X-Gitea-Debug: CacheControl=180s
```

These headers will make developers think more about: hey, the calls to AddCacheControlToHeader overwrite others ...

Until there is a complete fix and the conflict would get fully fixed one day .....

Reply:
I pushed the rename, can you take over?

Reply:
It looks clear to me, thank you ~~